### PR TITLE
Build SDK bindings for Flutter

### PR DESCRIPTION
**Pull request by dangeross**
This PR adds the ability to locally build the SDK bindings into the Flutter package of use when building Misty Breez (locally or with CI) against non-published SDK bindings. By default it is setup for local builds which are overridden during CI publishing.

- Adds scripts to build and copy SDK bindings (android / swift) using `just`
- Remove the CMakeLists recursive delete to prevent binding libraries from being deleted
- Add gradle.build.production to override gradle.build (local build) during publish
- Add a breez_sdk_liquid podspec which can be referenced during the Misty Breez non-published build

Misty Breez PR: https://github.com/breez/misty-breez/pull/190

### CODE REVIEW

File: lib/bindings/langs/flutter/justfile [hunk 1]
```diff
@@ -54,6 +55,16 @@ build-android profile='frb-min':
 build-other profile='frb-min':
 	dart scripts/build_other.dart --profile {{profile}}
 
+build-bindings:
```
Comment (erdemyerebasmaz) [inline lib/bindings/langs/flutter/justfile:58]: Can we rename 'bindings' to 'uniffi' in all occurrences?
```suggestion
build-bindings:
```

File: packages/flutter/android/build.gradle [hunk 2]
```diff
@@ -52,8 +56,22 @@ android {
...
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion 24
     }
 }
+
+dependencies {
+    implementation "net.java.dev.jna:jna:5.14.0@aar"
+    /* JSON serialization */
+    implementation 'org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.3'
```
Comment (erdemyerebasmaz) [inline packages/flutter/android/build.gradle:76]: Kotlin 1.8.20+ has serialization plugin. I don't think we need to depend on it explicitly.

File: packages/flutter/android/build.gradle [hunk 3]
```diff
@@ -4,6 +4,7 @@ group 'com.example.flutter_breez_liquid'
 version '0.2.1'
 
 buildscript {
+    ext.kotlin_version = '1.8.20'
```
Comment (erdemyerebasmaz) [inline packages/flutter/android/build.gradle:7]: If there are no obstacles, I'd suggest we upgrade to Kotlin 2.0.0.

File: packages/flutter/ios/breez_sdk_liquid.podspec [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# To learn more about a Podspec see http://guides.cocoapods.org/syntax/podspec.html.
+# Run `pod lib lint breez_sdk.podspec` to validate before publishing.
+Pod::Spec.new do |s|
+  s.name                   = 'BreezSDKLiquid'
+  s.version                = '0.2.1'
```
Comment (erdemyerebasmaz) [inline packages/flutter/ios/breez_sdk_liquid.podspec:5]: version script should update the version on this podspec.

File: packages/flutter/android/build.gradle.production [hunk 5]
```diff
@@ -0,0 +1,59 @@
+// The Android Gradle Plugin builds the native code with the Android NDK.
+
+group 'com.example.flutter_breez_liquid'
+version '0.2.1'
```
Comment (erdemyerebasmaz) [inline packages/flutter/android/build.gradle.production:4]: version script should update this version

File: packages/flutter/android/build.gradle (hunk 2 above)
Comment (dangeross) [reply packages/flutter/android/build.gradle:74]: Doesn't compile without this

File: packages/flutter/android/build.gradle (hunk 3 above)
Comment (dangeross) [reply packages/flutter/android/build.gradle:5]: I would prefer to keep this and the android bindings at the same kotlin version. If need be we can address it in a separate PR

File: lib/bindings/langs/flutter/justfile (hunk 1 above)
Comment (dangeross) [reply lib/bindings/langs/flutter/justfile]: Done [00bfede](https://github.com/breez/breez-sdk-liquid/pull/488/commits/00bfede5f84fbd40e966b444db74a14a88fa0695)

File: packages/flutter/ios/breez_sdk_liquid.podspec (hunk 4 above)
Comment (dangeross) [reply packages/flutter/ios/breez_sdk_liquid.podspec]: Done [00bfede](https://github.com/breez/breez-sdk-liquid/pull/488/commits/00bfede5f84fbd40e966b444db74a14a88fa0695)

File: packages/flutter/android/build.gradle.production (hunk 5 above)
Comment (dangeross) [reply packages/flutter/android/build.gradle.production]: Done [00bfede](https://github.com/breez/breez-sdk-liquid/pull/488/commits/00bfede5f84fbd40e966b444db74a14a88fa0695)

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM!!

**Pull request merged**